mir_eval/hierarchy.py

```diff
@@ -210,29 +298,22 @@ def _gauc(ref_lca, est_lca, transitive, window):
         ref_score = np.asarray(ref_score.todense()).squeeze()
         est_score = np.asarray(est_score.todense()).squeeze()
 
-        if transitive:
-            # Transitive: count comparisons across any level
-            ref_rank = np.greater.outer(ref_score, ref_score)
-        else:
-            # Non-transitive: count comparisons only across immediate levels
-            ref_rank = np.equal.outer(ref_score, ref_score + 1)
-
-        est_rank = np.greater.outer(est_score, est_score)
-
         # Don't count the query as a result
         # when query < window, query itself is the index within the slice
         # otherwise, query is located at the center of the slice, window
         # (this also holds when the slice goes off the end of the array.)
         idx = min(query, window)
-        ref_rank[idx, :] = False
-        ref_rank[:, idx] = False
 
-        # Compute normalization constant
-        normalizer = float(ref_rank.sum())
+        ref_score = np.concatenate((ref_score[:idx], ref_score[idx+1:]))
+        est_score = np.concatenate((est_score[:idx], est_score[idx+1:]))
 
-        # Add up agreement for frames
-        if normalizer > 0:
-            score += np.sum(np.logical_and(ref_rank, est_rank)) / normalizer
+        # If there are no comparisons to be made, move to the next frame
```

Comment:
This comment seems out of place. I don't think that's that what `_compare_frame_rankings` does. Also, I would remove the next empty line.